File: apps/desktop/src-tauri/src/app/clipboard.rs
use arboard::Clipboard;
use tauri::command;

#[command]
pub fn get_clipboard_html() -> Result<String, String> {
    let mut clipboard = Clipboard::new().map_err(|e| e.to_string())?;

    let html_data = clipboard.get().html();
    match html_data {
        Ok(data) => Ok(data),
        Err(_) => Err("Clipboard does not contain HTML data.".to_string()),
    }
}

#[command]
pub fn get_clipboard_text() -> Result<String, String> {
    let mut clipboard = Clipboard::new().map_err(|e| e.to_string())?;

    let text_data = clipboard.get_text();
    match text_data {
        Ok(data) => Ok(data),
        Err(_) => Err("Clipboard does not contain text data.".to_string()),
    }
}

File: apps/desktop/src/helper/clipboard.ts
import { invoke } from "@tauri-apps/api/core"
import { logger } from "./logger"

export const clipboardRead = async () => {
  let html = '',
    text = ''

  try {
    html = await invoke<string>('get_clipboard_html')
  } catch (error) {
    logger.error('get_clipboard_html error: ', error)
  }
  try {
    text = await invoke<string>('get_clipboard_text')
  } catch (error) {
    logger.error('get_clipboard_text error: ', error)
  }

  return {
    html,
    text,
  }
}
